kedro/extras/datasets/api/api_dataset.py

```diff
@@ -17,27 +17,26 @@ class APIDataSet(AbstractDataSet[None, requests.Response]):
     """``APIDataSet`` loads the data from HTTP(S) APIs.
     It uses the python requests library: https://requests.readthedocs.io/en/latest/
 
-    Example adding a catalog entry with
-    `YAML API
-    <https://kedro.readthedocs.io/en/stable/data/\
-        data_catalog.html#use-the-data-catalog-with-the-yaml-api>`_:
+    Example using the
+    `YAML API <https://kedro.readthedocs.io/en/stable/data/\
+    data_catalog.html#use-the-data-catalog-with-the-yaml-api>`_:
 
     .. code-block:: yaml
 
-        >>> usda:
-        >>>   type: api.APIDataSet
-        >>>   url: https://quickstats.nass.usda.gov
-        >>>   params:
-        >>>     key: SOME_TOKEN,
-        >>>     format: JSON,
-        >>>     commodity_desc: CORN,
-        >>>     statisticcat_des: YIELD,
-        >>>     agg_level_desc: STATE,
-        >>>     year: 2000
-        >>>
-
-
-    Example using Python API:
+        usda:
+          type: api.APIDataSet
+          url: https://quickstats.nass.usda.gov
+          params:
+            key: SOME_TOKEN,
+            format: JSON,
+            commodity_desc: CORN,
+            statisticcat_des: YIELD,
+            agg_level_desc: STATE,
+            year: 2000
+
+    Example using the
```

Comment:
If we're linking to the YAML API, should we not link to the code API docs?

Other possibility: rename the "code API" link to "Python API", since some could say YAML is also code.

Ideally, I think the examples should be equivalent, and our docs should show different options to achieving the same thing. Bit weird we build different datasets using YAML and Python syntax, as if the one you use should depend on your use case.

Reply:
Yeah, I think we should remove those examples from https://kedro.readthedocs.io/en/stable/data/data_catalog.html#use-the-data-catalog-with-the-yaml-api and put them in the API docs actually. I don't think we use API docs sufficiently for code examples at times and it makes the written markdown quite verbose at times. We could give the examples in docstrings and build them as examples within API docs.